### PR TITLE
about colors in surface plots

### PR DESCRIPTION
**Pull request by fchapoton**
here is the first step: distinguish 2 cases

then one need to know what format should be send to the next step in the toolchain

could you remind me where to find that ?

### CODE REVIEW

**Comment by williamstein** [comment]
Thanks for working on this!

**Comment by fchapoton** [comment]
Well, this is not yet done, and may/will require more work. In particular I may need to change also
smc/src/smc-webapp/3d.coffee

And doing this without being able to test will certainly be difficult..

**Comment by fchapoton** [comment]
I think I got a good way to do that, but I cannot test it.

**Comment by fchapoton** [comment]
now this branch needs testing, please.

**Comment by williamstein** [comment]
Thanks!  I will test it tomorrow (too tired now).

File: src/smc-webapp/3d.coffee [hunk 1]
```diff
@@ -428,6 +428,11 @@ class SalvusThreeJS
     add_obj: (myobj)=>
         @show_canvas()
 
+        if myobj.type == "index_face_set"
+            has_local_colors = myobj.has_local_colors
+        else
+            hase_local_colors = false
```
Comment (williamstein) [inline src/smc-webapp/3d.coffee]: typo: hase_ --> has

**Comment by fchapoton** [comment]
typo corrected.

**Comment by williamstein** [comment]
@fchapoton -- unfortunately this is still very broken.  Would you consider learning enough to actually test it?   I can pre-set up an SMC project for you for development...

**Comment by fchapoton** [comment]
damn.. I would really like this to work. Well, I have a SMC account already. Forgot under which email.

**Comment by williamstein** [comment]
I will set everything up for you here: https://cloud.sagemath.com/projects/8208a5a2-ac42-4b75-bf5c-ba7cf1eb89a4/files/, then explain exactly what you need to do to change things and test.

**Comment by fchapoton** [comment]
ok, thanks

**Comment by williamstein** [comment]
That project is now ready -- check out https://cloud.sagemath.com/projects/8208a5a2-ac42-4b75-bf5c-ba7cf1eb89a4/files/chat.sage-chat

**Comment by fchapoton** [comment]
I have some difficulties in acceding to the project right now. Trying to open the chat or to see 3d.coffee just keep loading / do nothing.

**Comment by haraldschilly** [comment]
so, the chat was broken up until a few minutes ago, and for me the project
loads and well, I don't see any problems with 3d.coffee... Just reload the
webpage to get the update for the chat. That should do it!

On Sat, Aug 20, 2016 at 9:38 PM, Frédéric Chapoton <notifications@github.com

**Comment by fchapoton** [comment]
Still the same problem for me: chat loading forever and nothing in 3D.coffee and other files.
Anyway my main problem is the fact that I cannot "pip install" as  William told me to do in the chat,
in order to be able to test my code..

**Comment by fchapoton** [comment]
so, what is happening about that now ?

**Comment by fchapoton** [comment]
this should be ready, once tested..

**Comment by williamstein** [comment]
Wow, it is incredibly hard to get this to work...  E.g., with the version of the code right now, most simple possible example fails: 
```
sphere()
```
results in 
```
...
  File "/projects/45f4aab5-7698-4ac8-9f63-9fd307401ad7/smc/src/data/projects/5b08ef88-9695-4377-91ab-1d09406afc6d/.local/lib/python2.7/site-packages/smc_sagews/graphics.py", line 327, in convert_index_face_set
    face_geometry = [{"material_name": p.texture.id,
AttributeError: 'NoneType' object has no attribute 'id'
```

**Comment by williamstein** [comment]
I also tried running the current live code on the backend and the frontend renderer gives this error `WARNING: rendering colored face with 2 vertices not implemented`

**Comment by fchapoton** [comment]
A suggestion for the p.texture.id problem, but I am not really convinced: could you try to add a line "p.triangulate()" before the line "p.obj()" ?

Faces with 2 vertices ?  Damn..

**Comment by williamstein** [comment]
OK, I tried a combination of things in the code and it seems to work now.

**Comment by fchapoton** [comment]
good. I am curious to see the changes.

**Comment by williamstein** [comment]
This is what I did -- now in master and will be live soon: https://github.com/sagemathinc/smc/commit/d9633ffdd80c219b04fab0d6865baaa17d8d26e1

**Pull request merged**